plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -436,61 +501,226 @@ fu_nordic_hid_cfg_channel_add_peers(FuNordicHidCfgChannel *self, GError **error)
 						0,
 						error))
 		return FALSE;
+
 	if (fu_nordic_hid_cfg_channel_cmd_receive(self,
 						  CONFIG_STATUS_DISCONNECTED,
 						  res,
 						  &error_local)) {
-		/* no peers */
+		/* forwarding configuration channel to peers not supported */
 		return TRUE;
 	}
 
 	/* Peers available */
 	if (!fu_nordic_hid_cfg_channel_cmd_receive(self, CONFIG_STATUS_SUCCESS, res, error))
 		return FALSE;
 
+	*cmd_supported = TRUE;
+	return TRUE;
+}
+
+static gboolean
+fu_nordic_hid_cfg_channel_get_next_peer_id_cmd(FuNordicHidCfgChannel *self,
+					       GError **error,
+					       guint8 *peer_id)
+{
+	g_autoptr(FuNordicCfgChannelMsg) res = g_new0(FuNordicCfgChannelMsg, 1);
+
+	if (!fu_nordic_hid_cfg_channel_cmd_send(self,
+						NULL,
+						NULL,
+						CONFIG_STATUS_GET_PEER,
+						NULL,
+						0,
+						error))
+		return FALSE;
+
+	if (!fu_nordic_hid_cfg_channel_cmd_receive(self, CONFIG_STATUS_SUCCESS, res, error))
+		return FALSE;
+
+	*peer_id = res->data[8];
+
+	return TRUE;
+}
+
+static gboolean
+fu_nordic_hid_cfg_channel_read_peers_cache_cmd(FuNordicHidCfgChannel *self,
+					       GError **error,
+					       gboolean *cmd_supported,
+					       guint8 peers_cache[PEERS_CACHE_LEN])
+{
+	g_autoptr(FuNordicCfgChannelMsg) res = g_new0(FuNordicCfgChannelMsg, 1);
+	g_autoptr(GError) error_local = NULL;
+
+	*cmd_supported = FALSE;
+
+	if (!fu_nordic_hid_cfg_channel_cmd_send(self,
+						NULL,
+						NULL,
+						CONFIG_STATUS_GET_PEERS_CACHE,
+						NULL,
+						0,
+						error))
+		return FALSE;
+
+	if (fu_nordic_hid_cfg_channel_cmd_receive(self,
+						  CONFIG_STATUS_DISCONNECTED,
+						  res,
+						  &error_local)) {
+		/* Configuration channel peers cache not supported. */
+		return TRUE;
+	}
+
+	/* Configuration channel peer caching available */
+	if (!fu_nordic_hid_cfg_channel_cmd_receive(self, CONFIG_STATUS_SUCCESS, res, error))
+		return FALSE;
+
+	if (!fu_memcpy_safe(peers_cache,
+			    PEERS_CACHE_LEN,
+			    0,
+			    res->data,
+			    PEERS_CACHE_LEN,
+			    0,
+			    PEERS_CACHE_LEN,
+			    error))
+		return FALSE;
+
+	*cmd_supported = TRUE;
+
+	return TRUE;
+}
+
+static gboolean
+fu_nordic_hid_cfg_channel_update_peers(FuNordicHidCfgChannel *self,
+				       GError **error,
+				       guint8 peers_cache[PEERS_CACHE_LEN])
+{
+	g_autoptr(GError) error_local = NULL;
```

Comment:
g_autoptr() usually come last in the variable block

Reply:
Aligned